### PR TITLE
Catch font errors

### LINKED ISSUE #1327: Hard crash when setting font with invalid size
## Steps to Reproduce
Since 96bdd4f9d883b5ff11d0409d9a24a6879db579d2, this will cause a hard crash from an empty MaybeLocal because parseFont returns undefined.

```js
const {createCanvas} = require(".");
const ctx = createCanvas(10,10).getContext("2d");
ctx.font = "bold undefinedpx Arial";
```

Crashing at
https://github.com/Automattic/node-canvas/blob/7de2441ee517a3681c5898a9a8520f4bed9619cc/src/CanvasRenderingContext2d.cc#L2455

## Your Environment
* Version of node-canvas (output of `npm list canvas` or `yarn list canvas`): 96bdd4f9d883b5ff11d0409d9a24a6879db579d2
* Environment (e.g. node 4.2.0 on Mac OS X 10.8): Win10, Node 10

### PR DESCRIPTION
**Pull request by Hakerh400**
Fix #1327

### CODE REVIEW

File: src/CanvasRenderingContext2d.cc [hunk 1]
```diff
@@ -2451,8 +2451,19 @@ NAN_SETTER(Context2d::SetFont) {
...
+  if (try_catch.HasCaught()){
+    try_catch.ReThrow();
+    return;
+  }
+  MaybeLocal<Object> maybeFont = Nan::To<Object>(maybeParsed.ToLocalChecked());
+  if (try_catch.HasCaught()){
+    try_catch.ReThrow();
+    return;
+  }
+  Local<Object> font = maybeFont.ToLocalChecked();
```
Comment (zbjornson) [inline src/CanvasRenderingContext2d.cc]: Thanks for the fast fix :).

A bit simpler: you can explicitly check if the `MaybeLocal` is empty with something like:

```cpp
  MaybeLocal<Object> mlfont = Nan::To<Object>(parsed);
  if (mlfont.IsEmpty()) return;
  Local<Object> font = mlfont.ToLocalChecked();
```

The error will still be thrown appropriately when the handle is empty.

Comment (Hakerh400) [reply src/CanvasRenderingContext2d.cc]: Done. Is there a need to test if `mparsed` is empty (can `parseFont` itself throw an error)?

Comment (zbjornson) [reply src/CanvasRenderingContext2d.cc]: Technically I'm not sure if there's any v8 operation that returns a `MaybeLocal` that is guaranteed not to fail. I'm not sure how unwise it is to assume simple operations don't fail. Having it doesn't hurt.
